app/src/main/java/org/groundplatform/android/persistence/remote/firebase/schema/SurveysCollectionReference.kt

```diff
@@ -36,10 +37,17 @@ class SurveysCollectionReference internal constructor(ref: CollectionReference)
 
   fun getReadable(user: User): Flow<List<Survey>> =
     reference()
+      .whereEqualTo(GENERAL_ACCESS_FIELD, SurveyProto.GeneralAccess.RESTRICTED_VALUE)
       .whereIn(
         FieldPath.of(ACL_FIELD, user.email),
         listOf(Role.SURVEY_ORGANIZER, Role.DATA_COLLECTOR, Role.VIEWER).map { it.ordinal },
       )
       .snapshots()
       .map { it.documents.map { doc -> doc.let { SurveyConverter.toSurvey(doc) } } }
+
+  fun getPublicReadable(): Flow<List<Survey>> =
+    reference()
+      .whereEqualTo(GENERAL_ACCESS_FIELD, SurveyProto.GeneralAccess.PUBLIC_VALUE)
+      .snapshots()
+      .map { it.documents.map { doc -> doc.let { SurveyConverter.toSurvey(doc) } } }
```

Comment:
Can doc really be null here? If so I'd use `documents.mapNotNull` instead of `map`+ `let`.